### ISSUE TITLE
SUBMIT 'index.android.bundle' missing when submitting to Playstore

### ISSUE THREAD

**Issue by xD3CODER** [expo/eas-cli#908]
### Build/Submit details page URL

https://expo.dev/accounts/urbendo/projects/urbendo/submissions/f4df9959-53a4-4b2d-b82d-c0c2028c3560

### Summary

When I submit the app to Playstore using eas cli, the download works fine.
But the app crashes immediately on startup.

I have seen a similar error and it seems to be related to Fastlane.

### Managed or bare?

bare

### Environment

EAS CLI 0.42.4 environment info:
System:
OS: Linux 5.13 Ubuntu 21.10 21.10 (Impish Indri)
Shell: 5.8 - /usr/bin/zsh
Binaries:
Node: 14.18.1 - ~/.nvm/versions/node/v14.18.1/bin/node
Yarn: 1.22.17 - ~/.nvm/versions/node/v14.18.1/bin/yarn
npm: 8.3.0 - ~/.nvm/versions/node/v14.18.1/bin/npm
Utilities:
Git: 2.32.0 - /usr/bin/git
npmPackages:
expo: ^44.0.3 => 44.0.3
react: 17.0.1 => 17.0.1
react-native: 0.64.3 => 0.64.3
npmGlobalPackages:
eas-cli: 0.42.4
expo-cli: 5.0.3
Project workflow: generic

### Error output

FATAL EXCEPTION: create_react_context
Process: com.app, PID: 24350
java.lang.RuntimeException: Unable to load script. Make sure you're either running Metro (run 'react-native start') or that your bundle 'index.android.bundle' is packaged correctly for release.
	at com.facebook.react.bridge.CatalystInstanceImpl.jniLoadScriptFromAssets(Native Method)
	at com.facebook.react.bridge.CatalystInstanceImpl.loadScriptFromAssets(CatalystInstanceImpl.java:2)
	at com.facebook.react.bridge.JSBundleLoader$1.loadScript(JSBundleLoader.java:1)
	at com.facebook.react.bridge.CatalystInstanceImpl.runJSBundle(CatalystInstanceImpl.java:3)
	at com.facebook.react.n.s(ReactInstanceManager.java:37)
	at com.facebook.react.n.c(ReactInstanceManager.java:1)
	at com.facebook.react.n$f.run(ReactInstanceManager.java:12)
	at java.lang.Thread.run(Thread.java:923)

### Reproducible demo or steps to reproduce from a blank project

Generate app-bundle using `eas build --platform android --profile=production --clear-cache --local `

**Comment by xD3CODER** [comment]
I just checked the signed apk downloaded from the playstore and indeed `index.android.bundle` is missing in the `assets` folder

**Comment by xD3CODER** [comment]
The file is missing only when using `app-bundle` build type
If I generate `.aab` manually using `./gradlew bundleRelease` the `index.android.bundle` is present in the `assets` folder

**Comment by xD3CODER** [comment]
It was not related to eas but to this [issue](https://github.com/facebook/react-native/issues/31157)

**Comment by Slapbox** [comment]
@xD3CODER did you actually get this fixed? We're hitting this issue when building development builds.

**Comment by samvoults** [comment]
Any news? We are stuck to publish our new release too. I've read some threads where people used a previous gradle version to fix the problem here: https://stackoverflow.com/questions/72564710/react-native-crash-release-build-fatal-exception-create-react-context

"The Android Gradle Plugin version was 7.0.4 when the project was first created. Android Studio version asked me to upgrade to 7.2.1, so I upgraded. It worked when I downgraded the version to 7.0.4.

android>build.gradle

classpath('com.android.tools.build:gradle:7.0.4')"

Can we do the same or the file "index.android.bundle" is handled by expo?
